helix-stdx/src/env.rs

```diff
@@ -30,6 +31,41 @@ pub fn set_current_working_dir(path: impl AsRef<Path>) -> std::io::Result<()> {
     Ok(())
 }
 
+pub fn env_var_is_set(env_var_name: &str) -> bool {
+    std::env::var_os(env_var_name).is_some()
+}
+
+pub fn binary_exists<T: AsRef<OsStr>>(binary_name: T) -> bool {
+    which::which(binary_name).is_ok()
+}
+
+pub fn which<T: AsRef<OsStr>>(
+    binary_name: T,
+) -> Result<std::path::PathBuf, ExecutableNotFoundError> {
+    which::which(binary_name.as_ref()).map_err(|err| ExecutableNotFoundError {
+        command: binary_name.as_ref().to_string_lossy().into_owned(),
+        inner: err,
+    })
+}
+
+#[derive(Debug)]
+pub struct ExecutableNotFoundError {
+    command: String,
+    inner: which::Error,
+}
+
+impl std::fmt::Display for ExecutableNotFoundError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "could not find '{}' executable: {}",
```

Comment:
```suggestion
            "command '{}' not found: {}",
```

nit: I think this sounds better (closer to the error messages in shells) :D